src/pymor/operators/interface.py

```diff
@@ -520,35 +520,60 @@ def restricted(self, dofs):
         """
         raise NotImplementedError
 
-    def __add__(self, other):
-        """Sum of two operators."""
-        if other == 0:
-            return self
+    def _add_sub(self, other, sign):
         if not isinstance(other, Operator):
             return NotImplemented
         from pymor.operators.constructions import LincombOperator
-        if isinstance(other, LincombOperator):
-            return NotImplemented
+        if self.name != 'LincombOperator' or (self.name == 'LincombOperator'
+                                              and not isinstance(self, LincombOperator)):
+            if other.name == 'LincombOperator' and isinstance(other, LincombOperator):
+                operators = (self,) + other.operators
+                coefficients = (1.,) + (other.coefficients if sign == 1. else tuple(-c for c in other.coefficients))
+            else:
+                operators, coefficients = (self, other), (1., sign)
+        elif other.name == 'LincombOperator' and isinstance(other, LincombOperator):
+            operators = self.operators + other.operators
+            coefficients = self.coefficients + (other.coefficients if sign == 1.
+                                                else tuple(-c for c in other.coefficients))
         else:
-            return LincombOperator([self, other], [1., 1.])
+            operators, coefficients = self.operators + (other,), self.coefficients + (sign,)
 
-    __radd__ = __add__
+        return LincombOperator(operators, coefficients, solver_options=self.solver_options)
 
-    def __sub__(self, other):
+    def _radd_sub(self, other, sign):
         if not isinstance(other, Operator):
             return NotImplemented
```

Comment:
I think the only possibility now that this method can be called is when `other` is not an `Operator` and we only need to take care of the case that `other` might be `0`.

Reply:
right